incubator/jaeger/templates/common-cm.yaml

```diff
@@ -0,0 +1,83 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ template "jaeger.fullname" . }}
+  labels:
+    app: {{ template "jaeger.name" . }}
+    jaeger-infra: common-configmap
+    chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}
+    heritage: {{ .Release.Service }}
+    release: {{ .Release.Name }}
+data:
+  cassandra.contact-points: {{ template "cassandra.contact_points" . }}
+  cassandra.datacenter.name: {{ .Values.cassandra.config.dc_name | quote }}
+  cassandra.keyspace: {{ printf "%s_%s" "jaeger_v1" .Values.cassandra.config.dc_name | quote }}
+  cassandra.schema.mode: {{ .Values.schema.mode | quote }}
+  cassandra.servers: {{ template "cassandra.fullname" . }}
+  collector.host-port: {{ template "jaeger.collector.host-port" . }}
+  collector.http-port: {{ .Values.collector.service.httpPort | quote }}
+  collector.port: {{ .Values.collector.service.tchannelPort | quote }}
+  collector.zipkin.http-port: {{ .Values.collector.service.zipkinPort | quote }}
+  es.server-urls: {{ template "elasticsearch.client.url" . }}
```

Comment:
This forces the external or previously deployed elasticsearch host to be <relname>-elasticsearch. This should be set to  `.Values.storage.elasticsearch.host`.

Reply:
What values are you using it with? I have tested this with both external and provisioned ES and it is working fine for me. Take a closer look at this helper:
```
{{- define "elasticsearch.client.url" -}}
{{- $port := .Values.storage.elasticsearch.port | toString -}}
{{- if .Values.provisionDataStore.elasticsearch -}}
{{- $host := printf "%s-%s-%s" .Release.Name "elasticsearch" "client" | trunc 63 | trimSuffix "-" -}}
{{- printf "%s://%s:%s" .Values.storage.elasticsearch.scheme $host $port }}
{{- else }}
{{- printf "%s://%s:%s" .Values.storage.elasticsearch.scheme .Values.storage.elasticsearch.host $port }}
{{- end -}}
{{- end -}}
```
It doesn't force anything, right?

Reply:
Sorry.... small goof. I got hit by the `storage.type` thing again by not setting it. I corrected that now but it looks like the collector and query components are having problems connecting to elasticsearch.

These are the command line parameters I am running with:
```
--name myrel --set provisionDataStore.cassandra=false --set provisionDataStore.elasticsearch=false --set storage.type=elasticsearch --set storage.elasticsearch.host=elasticsearch --set storage.elasticsearch.port=9200 --set storage.elasticsearch.user=elastic --set storage.elasticsearch.password=changeme
```

Get pods output:
```
[dev@k8controller ~]$ kubectl get pods
NAME                                     READY     STATUS             RESTARTS   AGE
elasticsearch-0                          1/1       Running            0          24m
elasticsearch-1                          1/1       Running            0          24m
elasticsearch-2                          1/1       Running            0          24m
myrel-jaeger-agent-31jxv                 1/1       Running            0          8m
myrel-jaeger-agent-89kxf                 1/1       Running            0          8m
myrel-jaeger-agent-gldbw                 1/1       Running            0          8m
myrel-jaeger-agent-lh8tx                 1/1       Running            0          8m
myrel-jaeger-agent-t7c0h                 1/1       Running            0          8m
myrel-jaeger-agent-wgvrm                 1/1       Running            0          8m
myrel-jaeger-collector-623630575-b32fr   0/1       CrashLoopBackOff   6          8m
myrel-jaeger-query-3159756439-nk7c6      0/1       CrashLoopBackOff   6          8m
```

Error:
```
{"level":"fatal","ts":1515108228.394109,"caller":"collector/main.go:99","msg":"Unable to set up builder","error":"health check timeout: no Elasticsearch node available"
```

Verified the configmap has the correct value:
```
Name:		myrel-jaeger
Namespace:	default
Labels:		app=jaeger
		chart=jaeger-0.3.0
		heritage=Tiller
		jaeger-infra=common-configmap
		release=myrel
Annotations:	<none>

Data
====
...
es.server-urls:
----
http://elasticsearch:9200
...
```

Reply:
Oh, good catch. I've tested with ES without username and password. I'll add those two if they are set. As you can see in this file on lines 60 and 62 they haven't been implemented.

Reply:
@pavelnikolov I would also recommend setting the default username and password to elastic/changeme since those are the defaults.